lib/policy_machine_storage_adapters/active_record/postgresql.rb

```diff
@@ -2,6 +2,95 @@
 
 module PolicyMachineStorageAdapter
   class ActiveRecord
+    class PolicyElementAssociation
+      def self.all_accessible_objects(associations, root_id: nil)
```

Comment:
I'm a little skeptical that a sql method that does two recursive functions is going to be faster on big data. I'd be curious about performance on that

Reply:
very fair, i'll do some sandbox performance testing with highly assigned users and update when i get more data